Fix Spark version build problems

jthelin opened the pull request:
(1) Fix outdated version of Spark binary distributions from Apache being used in Mobius builds.
- Specifically, Spark `2.0.0` is no longer available as a binary `.tgz` distribution from Apache Software Foundation Distribution server.
  http://www.us.apache.org/dist/spark/
- Only versions `1.6.2` and `2.0.1` of Spark binary distributions are available for download from the Apache Distribution server now.

```
http://www.us.apache.org/dist/spark/

[DIR] spark-1.6.2/            2016-06-25 07:38    -
[DIR] spark-2.0.1/            2016-10-04 17:26    -
```
- This Pull Request changes the Spark version defaults in Mobius download scripts to match what is the currently available from Apache:
  
  `1.6.1` -> `1.6.2` and `2.0.0` -> `2.0.1`

(2) Add guard code to check that Download-File function got a real file.
- Check in the `Download-File` function that the current downloaded copy of the file is not empty [zero length].

**Failure Mode**
[From my VSO hosted build server environment - which does no caching of previous downloaded files]

```
2016-10-27T16:30:29.7363825Z [downloadtools.Download-File] Start downloading http://www.us.apache.org/dist/spark/spark-2.0.0/spark-2.0.0-bin-hadoop2.6.tgz to C:\a\1\s\build\tools\spark-2.0.0-bin-hadoop2.6.tgz ...
2016-10-27T16:30:29.7763815Z 
2016-10-27T16:30:29.7813808Z Id     Name            PSJobTypeName   State         HasMoreData     Location  
2016-10-27T16:30:29.7813808Z --     ----            -------------   -----         -----------     --------  
2016-10-27T16:30:29.7883809Z 1      Web.Download...                 NotStarted    False                     
2016-10-27T16:30:29.7913858Z 2      Web.Download...                 NotStarted    False                     
2016-10-27T16:30:29.9483826Z [downloadtools.Download-File] Download completed. Time taken: 202 milliseconds
2016-10-27T16:30:29.9513819Z [downloadtools.Untar-File] Extracting C:\a\1\s\build\localmode\..\tools\spark-2.0.0-bin-hadoop2.6.tgz to C:\a\1\s\build\localmode\..\tools ...
2016-10-27T16:30:30.0603812Z Error extracting C:\a\1\s\build\localmode\..\tools\spark-2.0.0-bin-hadoop2.6.tgz 
2016-10-27T16:30:30.0603812Z Exception message EOS reading GZIP header
2016-10-27T16:30:30.0623815Z Usage : >TarTool sourceFile destinationDirectory
2016-10-27T16:30:30.0623815Z >TarTool D:\sample.tar.gz ./
2016-10-27T16:30:30.0623815Z >TarTool sample.tgz temp
2016-10-27T16:30:30.0623815Z >TarTool -x sample.tar temp
2016-10-27T16:30:30.6098995Z [downloadtools.Untar-File] Extraction completed. Time taken: 658 milliseconds
```

skaarthik commented:
Thanks for catching this, @jthelin 
Can you use http://archive.apache.org/dist/spark/spark-2.0.0/spark-2.0.0-bin-hadoop2.6.tgz instead of updating to 2.0.1?

jthelin commented:
Good idea to minimize version changes - thanks for finding the alternate location @skaarthik 

I switched the versions back to original and changed the server location to be `archive.apache.org` as suggested.

jthelin commented:
FYI, the Windows server build failed. I am investigating fix.

jthelin commented:
Changes are good-to-go now @skaarthik if you would like to take another look.
